pkg/cloud/services/provider/provider.go

```diff
@@ -67,6 +72,11 @@ func NewClientFromCluster(ctx context.Context, ctrlClient client.Client, openSta
 			return nil, nil, "", err
 		}
 	}
+
+	if caCert == nil && defaultCACert != nil {
```

Comment:
Considerung `defaultCACert == nil` and  `caCert == nil`, it shouldn't matter if we set `caCert = nil` or leave it `nil`.
```suggestion
	if caCert == nil {
```